org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -667,30 +668,38 @@ public  boolean render(Component content, MDashboardContent dc, DashboardRunnabl
     	int PA_Goal_ID = dc.getPA_Goal_ID();
     	if(PA_Goal_ID > 0)
     	{
-    		//link to open performance detail
-    		Div div = new Div();
-    		Toolbarbutton link = new Toolbarbutton();
-    		if (ThemeManager.isUseFontIconForImage())
-    			link.setIconSclass("z-icon-Zoom");
-    		else
-    			link.setImage(ThemeManager.getThemeResource("images/Zoom16.png"));
-            link.setAttribute("PA_Goal_ID", PA_Goal_ID);
-            link.addEventListener(Events.ON_CLICK, new EventListener<Event>() {
-				public void onEvent(Event event) throws Exception {
-					int PA_Goal_ID = (Integer)event.getTarget().getAttribute("PA_Goal_ID");
-					MGoal goal = new MGoal(Env.getCtx(), PA_Goal_ID, null);
-					new WPerformanceDetail(goal);
-				}
-            });
-            div.appendChild(link);
-            content.appendChild(div);
 
             String goalDisplay = dc.getGoalDisplay();
             MGoal goal = new MGoal(Env.getCtx(), PA_Goal_ID, null);
-            WGraph graph = new WGraph(goal, 55, false, true,
-            		!(MDashboardContent.GOALDISPLAY_Chart.equals(goalDisplay)),
-            		MDashboardContent.GOALDISPLAY_Chart.equals(goalDisplay));
-            content.appendChild(graph);
+            if(MDashboardContent.GOALDISPLAY_Goal.equals(goalDisplay)) {
+            	WPAWidget paWidget = new WPAWidget(goal, null);
+            	((HtmlBasedComponent)content).setSclass("performance-gadget");
```

Comment:
the cast to HtmlBasedComponent is not needed here.

Reply:
I'm not sure how can I set the css class without the cast? Variable _content_ is type of _Component_ and does not have a _setSclass()_ method.

Reply:
ok, my mistake here.